wasm/jit/engine.go

```diff
@@ -251,16 +251,31 @@ type compiledWasmFunction struct {
 	codeInitialAddress uintptr
 	// The same purpose as codeInitialAddress, but for memory.Buffer.
 	memoryAddress uintptr
+	// The max of the stack height this function can reach.
+	maxtStackHeight uint64
 }
 
 const (
 	builtinFunctionIndexGrowMemory = iota
 )
 
-func (e *engine) stackGrow() {
-	newStack := make([]uint64, len(e.stack)*2)
-	copy(newStack[:len(e.stack)], e.stack)
-	e.stack = newStack
+// Grow the stack size according to requiredHeight argument
+// which is the required height from the base pointer
+// for the next function frame execution.
+func (e *engine) maybeGrowStack(requiredHeight uint64) {
+	currentLen := uint64(len(e.stack))
+	remained := currentLen - e.currentBaseStackPointer
+	if requiredHeight > remained {
+		// This case we need to grow the stack as the empty slots
+		// are not able to store all the stack items.
+		// So we grow the stack with the new len = currentLen*2+required.
+		required := requiredHeight - remained
+		newStack := make([]uint64, currentLen*2+(required))
```

Comment:
hard to do math in my head, but will `currentLen*2+(required)` always be less than `requiredHeight`? if not, feels even less like a max :D

Reply:
right, fixed: https://github.com/tetratelabs/wazero/pull/68/commits/49c0763bfac7bf66ff7a4f6549c6e59e61fb37f7 :D